state/cloud.go

```diff
@@ -203,6 +237,23 @@ func (st *State) AddCloud(c cloud.Cloud, owner string) error {
 	return nil
 }
 
+// UpdateCloud updates an existing cloud with the given name and details.
+// Note that the Config is deliberately ignored - it's only
+// relevant when bootstrapping.
+func (st *State) UpdateCloud(c cloud.Cloud) error {
+	if err := validateCloud(c); err != nil {
+		return errors.Annotate(err, "invalid cloud")
```

Comment:
if validateCloud() error is comprehensive, wrapping with "invalid cloud" doesn't really add much